add GT4Py as example project using dynamic-image-name (#197)
eth-cscs/cscs-docs: docs/services/cicd.md

There is also a building block in the templates, name `.dynamic-image-name`, which you can use to get rid for most of the boilerplate.
It is important to note that this building block will export the dynamic name under the hardcoded name `BASE_IMAGE` in the `dotenv` file.

> Additionally the variable `DOCKER_TAG`, which is the tag of the image is exported in the `dotenv` file.

Reviewer: To me the sentence reads a bit funny. What about:
```suggestion
The variable `DOCKER_TAG`, containing the tag of the image, is also exported in the `dotenv` file.
```
?

`build base` is using additionally the building block `.dynamic-image-name`, while `build software` is unchanged.
Have a look at the definition of the block `.dynamic-image-name` in the file [.ci-ext.yml](https://gitlab.com/cscs-ci/recipes/-/blob/master/templates/v2/.ci-ext.yml) for further notes.

!!! example "GT4Py example"

>     An example using `.dynamic-image-name` in action can be found in the [gt4py repository](https://github.com/GridTools/gt4py/tree/main/ci)

Reviewer: ```suggestion
    An example using `.dynamic-image-name` in action can be found in the [gt4py repository](https://github.com/GridTools/gt4py/tree/main/ci).
```
